
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../simplify-taxa/">
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.20">
    
    
      
        <title>screen - phu</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.e53b48f4.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/_mkdocstrings.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#screen" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="phu" class="md-header__button md-logo" aria-label="phu" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            phu
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              screen
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/camilogarciabotero/phu" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    camilogarciabotero/phu
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="phu" class="md-nav__button md-logo" aria-label="phu" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    phu
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/camilogarciabotero/phu" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    camilogarciabotero/phu
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Commands
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Commands
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cluster/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    cluster
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../simplify-taxa/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    simplify-taxa
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    screen
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    screen
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-does-it-do" class="md-nav__link">
    <span class="md-ellipsis">
      What does it do?
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#synopsis" class="md-nav__link">
    <span class="md-ellipsis">
      Synopsis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-it-works" class="md-nav__link">
    <span class="md-ellipsis">
      How it works
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-multiple-hmms" class="md-nav__link">
    <span class="md-ellipsis">
      Using Multiple HMMs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#target-data-extraction-and-hmm-building" class="md-nav__link">
    <span class="md-ellipsis">
      Target Data Extraction and HMM Building
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#understanding-your-results" class="md-nav__link">
    <span class="md-ellipsis">
      Understanding Your Results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#command-options" class="md-nav__link">
    <span class="md-ellipsis">
      Command Options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examples" class="md-nav__link">
    <span class="md-ellipsis">
      Examples
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-to-expect" class="md-nav__link">
    <span class="md-ellipsis">
      What to expect
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    <span class="md-ellipsis">
      Requirements
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-does-it-do" class="md-nav__link">
    <span class="md-ellipsis">
      What does it do?
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#synopsis" class="md-nav__link">
    <span class="md-ellipsis">
      Synopsis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-it-works" class="md-nav__link">
    <span class="md-ellipsis">
      How it works
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-multiple-hmms" class="md-nav__link">
    <span class="md-ellipsis">
      Using Multiple HMMs
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#target-data-extraction-and-hmm-building" class="md-nav__link">
    <span class="md-ellipsis">
      Target Data Extraction and HMM Building
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#understanding-your-results" class="md-nav__link">
    <span class="md-ellipsis">
      Understanding Your Results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#command-options" class="md-nav__link">
    <span class="md-ellipsis">
      Command Options
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#examples" class="md-nav__link">
    <span class="md-ellipsis">
      Examples
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-to-expect" class="md-nav__link">
    <span class="md-ellipsis">
      What to expect
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    <span class="md-ellipsis">
      Requirements
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="screen">screen<a class="headerlink" href="#screen" title="Permanent link">&para;</a></h1>
<h2 id="what-does-it-do">What does it do?<a class="headerlink" href="#what-does-it-do" title="Permanent link">&para;</a></h2>
<p>The <code>phu screen</code> command helps you find DNA contigs that contain specific protein families. It predicts proteins from your contigs, searches those proteins against Hidden Markov Model (HMM) profiles using <strong>pyHMMER</strong> (a fast Python implementation), and then selects contigs based on the matches and your combination/filtering rules. Think of it as a molecular search engine for pulling out contigs that contain the proteins you care about.</p>
<p>This is especially useful when you have metagenomic assemblies and want to pull out contigs that belong to viruses, or when you're looking for contigs that contain specific metabolic pathways. The tool now includes advanced features like building custom HMMs from your target proteins and specialized viral gene prediction.</p>
<h2 id="synopsis">Synopsis<a class="headerlink" href="#synopsis" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span><span class="o">[</span>INPUT_CONTIGS<span class="o">]</span><span class="w"> </span><span class="o">[</span>HMMS...<span class="o">]</span>
</code></pre></div>
<p><strong>Example:</strong>
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>your_contigs.fasta<span class="w"> </span>your_protein_family.hmm
</code></pre></div></p>
<p>This simple command will find all contigs in <code>your_contigs.fasta</code> that contain proteins matching <code>your_protein_family.hmm</code> and save them to a new file called <code>screened_contigs.fasta</code> in a folder named <code>phu-screen</code>.</p>
<h2 id="how-it-works">How it works<a class="headerlink" href="#how-it-works" title="Permanent link">&para;</a></h2>
<p>The screen command follows four main steps with several optional enhancements:</p>
<p><strong>First</strong>, it predicts all possible proteins from your DNA contigs using <strong>pyrodigal</strong> (for standard microbial genes) or <strong>pyrodigal-gv</strong> (for viral genes when available). This step translates your DNA sequences into protein sequences, creating names like <code>contig123|gene1</code>, <code>contig123|gene2</code>, etc. The tool automatically handles complex contig names with multiple <code>|</code> separators.</p>
<p><strong>Second</strong>, it searches the predicted proteins against your HMM profiles using <strong>pyHMMER</strong>, a fast in-memory Python implementation that eliminates the need for external HMMER binaries. Each HMM file is processed efficiently with native Python threading, and results maintain compatibility with standard HMMER formats.</p>
<p><strong>Third</strong>, it decides which contigs to keep based on the search results and your filtering criteria. This is where the "combine mode" logic becomes important if you're using multiple HMMs, distinguishing between HMMER "targets" (protein sequences) and HMM "models" (profiles that matched).</p>
<p><strong>Finally</strong>, it extracts the matching contigs from your original file and saves them to the output. The tool can also extract target proteins per model and build custom HMM profiles from those proteins for future use.</p>
<h2 id="using-multiple-hmms">Using Multiple HMMs<a class="headerlink" href="#using-multiple-hmms" title="Permanent link">&para;</a></h2>
<p>When you provide multiple HMM files, you need to decide how strict you want to be about matches. There are three ways to combine results:</p>
<p><strong>Any mode</strong> (the default) keeps contigs that match at least one model. Important detail: when a single contig matches multiple models, "any" preserves the best hit per model (rather than selecting only one overall best hit). As a result a contig that matches model A and model B will yield one protein for A and one protein for B (subject to <code>--top-per-contig</code>). This is useful when you want a representative protein per matched model from each contig.</p>
<p><strong>All mode</strong> only keeps contigs that match every single HMM you provided. This is very strict and useful when you need complete sets of proteins. For instance, if you're looking for complete viral genomes that must have all four proteins (capsid, portal, primase, and terminase), you would use "all" mode.</p>
<p><strong>Threshold mode</strong> lets you specify a minimum number of models that must match. This gives you flexibility between "any" and "all". You might require at least 3 out of 5 models to match, for example.</p>
<h2 id="target-data-extraction-and-hmm-building">Target Data Extraction and HMM Building<a class="headerlink" href="#target-data-extraction-and-hmm-building" title="Permanent link">&para;</a></h2>
<p>The tool offers powerful features for analyzing and reusing your screening results:</p>
<p><strong>Target Protein Extraction</strong> (<code>--save-target-proteins</code>) saves the actual protein sequences that matched each HMM model, organized in separate files per model. These proteins come only from contigs that passed your final filtering criteria.</p>
<p><strong>Custom HMM Building</strong> (<code>--save-target-hmms</code>) automatically builds new HMM profiles from your target proteins. This works independently of protein saving - if you only want HMMs, the tool will extract proteins temporarily, build the HMMs, then clean up. For single sequences, it builds individual HMMs; for multiple sequences, it creates simple alignments by padding to equal length before building consensus models.</p>
<p><strong>Viral Mode Support</strong> - When using viral gene prediction (if pyrodigal-gv is available), the tool is optimized for shorter, more compact viral genes and can handle overlapping gene structures common in viral genomes.</p>
<h2 id="understanding-your-results">Understanding Your Results<a class="headerlink" href="#understanding-your-results" title="Permanent link">&para;</a></h2>
<p>The main output is <code>screened_contigs.fasta</code>, which contains all the contigs that passed your filtering criteria. You'll also get <code>kept_contigs.txt</code> with just the names of these contigs.</p>
<p>If you used multiple HMMs, pay attention to how the combine mode affects your results. In "any" mode (see above), a contig that matches multiple models will produce one (best) protein per matched model. In "all" mode, kept contigs must have at least one hit for every model, and the tool returns one best hit per model for each kept contig — so the per-model protein counts will be balanced across models.</p>
<p>When you use the <code>--save-target-proteins</code> option, you'll get a folder called <code>target_proteins</code> with separate files for each model. Note the distinction:</p>
<ul>
<li>In "pure" HMM mode (default), each input HMM file is treated as one model and the output files are grouped by HMM filename (one file per input file).</li>
<li>In "mixed" HMM mode (used for concatenated/pressed HMM files), a single HMM file can contain multiple model names; in that case <code>--save-target-proteins</code> will create one output file per model name found inside the domtblout.</li>
</ul>
<p>All saved protein FASTA files contain only proteins that come from contigs that were kept in the final <code>screened_contigs.fasta</code>, and are de-duplicated per model file.</p>
<p><strong>New Target Data Outputs:</strong>
- <code>target_proteins/{model}_proteins.mfa</code> - Proteins matching each model (if <code>--save-target-proteins</code>)
- <code>target_hmms/{model}.hmm</code> - Custom HMMs built from target proteins (if <code>--save-target-hmms</code>)</p>
<p>These outputs respect your HMM mode settings and combination logic, ensuring consistency between your screening criteria and extracted data.</p>
<h2 id="command-options">Command Options<a class="headerlink" href="#command-options" title="Permanent link">&para;</a></h2>
<div class="highlight"><pre><span></span><code>Usage:<span class="w"> </span>phu<span class="w"> </span>screen<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>HMMS...<span class="w">                                                        </span>

<span class="w"> </span>Screen<span class="w"> </span>contigs<span class="w"> </span><span class="k">for</span><span class="w"> </span>protein<span class="w"> </span>families<span class="w"> </span>using<span class="w"> </span>pyHMMER<span class="w"> </span>on<span class="w"> </span>predicted<span class="w"> </span>CDS.<span class="w">                          </span>

<span class="w"> </span>Supports<span class="w"> </span>multiple<span class="w"> </span>HMM<span class="w"> </span>files<span class="w"> </span>with<span class="w"> </span>different<span class="w"> </span>combination<span class="w"> </span>modes:<span class="w">                              </span>
<span class="w"> </span>-<span class="w"> </span>any:<span class="w"> </span>Keep<span class="w"> </span>contigs<span class="w"> </span>matching<span class="w"> </span>any<span class="w"> </span>HMM<span class="w"> </span><span class="o">(</span>default,<span class="w"> </span>most<span class="w"> </span>permissive<span class="o">)</span><span class="w">                            </span>
<span class="w"> </span>-<span class="w"> </span>all:<span class="w"> </span>Keep<span class="w"> </span>contigs<span class="w"> </span>matching<span class="w"> </span>all<span class="w"> </span>HMMs<span class="w"> </span><span class="o">(</span>most<span class="w"> </span>restrictive<span class="o">)</span><span class="w">                                   </span>
<span class="w"> </span>-<span class="w"> </span>threshold:<span class="w"> </span>Keep<span class="w"> </span>contigs<span class="w"> </span>matching<span class="w"> </span>at<span class="w"> </span>least<span class="w"> </span>--min-hmm-hits<span class="w"> </span>HMMs<span class="w">                            </span>

<span class="w"> </span>HMM<span class="w"> </span>modes:<span class="w">                                                                                 </span>
<span class="w"> </span>-<span class="w"> </span>pure:<span class="w"> </span>Each<span class="w"> </span>HMM<span class="w"> </span>file<span class="w"> </span>contains<span class="w"> </span>one<span class="w"> </span>model<span class="w"> </span><span class="o">(</span>default,<span class="w"> </span>most<span class="w"> </span>common<span class="o">)</span><span class="w">                            </span>
<span class="w"> </span>-<span class="w"> </span>mixed:<span class="w"> </span>HMM<span class="w"> </span>files<span class="w"> </span>contain<span class="w"> </span>multiple<span class="w"> </span>models<span class="w"> </span><span class="o">(</span>pressed/concatenated<span class="w"> </span>HMMs<span class="o">)</span><span class="w">                     </span>

<span class="w"> </span>Examples:<span class="w">                                                                                  </span>
<span class="w">   </span>phu<span class="w"> </span>screen<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--combine-mode<span class="w"> </span>any<span class="w"> </span>*.hmm
<span class="w">   </span>phu<span class="w"> </span>screen<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--combine-mode<span class="w"> </span>all<span class="w"> </span>file1.hmm<span class="w"> </span>file2.hmm<span class="w"> </span>file3.hmm
<span class="w">   </span>phu<span class="w"> </span>screen<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--combine-mode<span class="w"> </span>threshold<span class="w"> </span>--min-hmm-hits<span class="w"> </span><span class="m">5</span><span class="w"> </span>pfam_database.hmm
<span class="w">   </span>phu<span class="w"> </span>screen<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--save-target-proteins<span class="w"> </span>*.hmm
<span class="w">   </span>phu<span class="w"> </span>screen<span class="w"> </span>-i<span class="w"> </span>contigs.fa<span class="w"> </span>--save-target-hmms<span class="w"> </span>*.hmm

╭─<span class="w"> </span>Arguments<span class="w"> </span>──────────────────────────────────────────────────────────────────────────────╮
│<span class="w"> </span>*<span class="w">    </span>hmms<span class="w">      </span>HMMS...<span class="w">  </span>HMM<span class="w"> </span>files<span class="w"> </span><span class="o">(</span>supports<span class="w"> </span>wildcards<span class="w"> </span>like<span class="w"> </span>*.hmm<span class="o">)</span><span class="w"> </span><span class="o">[</span>required<span class="o">]</span><span class="w">             </span>│
╰──────────────────────────────────────────────────────────────────────────────────────────╯
╭─<span class="w"> </span>Options<span class="w"> </span>────────────────────────────────────────────────────────────────────────────────╮
│<span class="w"> </span>*<span class="w">  </span>--input-contigs<span class="w">     </span>-i<span class="w">                        </span>PATH<span class="w">                </span>Input<span class="w"> </span>contigs<span class="w"> </span>FASTA<span class="w"> </span>│
│<span class="w">                                                                      </span><span class="o">[</span>required<span class="o">]</span><span class="w">          </span>│
│<span class="w">    </span>--output-folder<span class="w">     </span>-o<span class="w">                        </span>PATH<span class="w">                </span>Output<span class="w"> </span>directory<span class="w">    </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w">           </span>│
│<span class="w">                                                                      </span>phu-screen<span class="o">]</span><span class="w">         </span>│
│<span class="w">    </span>--mode<span class="w">                                        </span>TEXT<span class="w">                </span>pyrodigal<span class="w"> </span>mode:<span class="w">     </span>│
│<span class="w">                                                                      </span>meta<span class="p">|</span>single<span class="w">         </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span>meta<span class="o">]</span><span class="w">     </span>│
│<span class="w">    </span>--threads<span class="w">           </span>-t<span class="w">                        </span>INTEGER<span class="w"> </span>RANGE<span class="w">       </span>Threads<span class="w"> </span><span class="k">for</span><span class="w"> </span>both<span class="w">    </span>│
│<span class="w">                                                  </span><span class="o">[</span>x&gt;<span class="o">=</span><span class="m">1</span><span class="o">]</span><span class="w">              </span>pyrodigal<span class="w"> </span>and<span class="w">       </span>│
│<span class="w">                                                                      </span>pyHMMER<span class="w">             </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span><span class="m">1</span><span class="o">]</span><span class="w">        </span>│
│<span class="w">    </span>--min-bitscore<span class="w">                                </span>FLOAT<span class="w">               </span>Minimum<span class="w"> </span>bitscore<span class="w"> </span>to<span class="w"> </span>│
│<span class="w">                                                                      </span>keep<span class="w"> </span>a<span class="w"> </span>domain<span class="w"> </span>hit<span class="w">   </span>│
│<span class="w">    </span>--max-evalue<span class="w">                                  </span>FLOAT<span class="w">               </span>Maximum<span class="w"> </span>independent<span class="w"> </span>│
│<span class="w">                                                                      </span>E-value<span class="w"> </span>to<span class="w"> </span>keep<span class="w"> </span>a<span class="w">   </span>│
│<span class="w">                                                                      </span>domain<span class="w"> </span>hit<span class="w">          </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span>1e-05<span class="o">]</span><span class="w">    </span>│
│<span class="w">    </span>--top-per-contig<span class="w">                              </span>INTEGER<span class="w">             </span>Keep<span class="w"> </span>top-N<span class="w"> </span>hits<span class="w"> </span>per<span class="w"> </span>│
│<span class="w">                                                                      </span>contig<span class="w"> </span><span class="o">(</span>by<span class="w">          </span>│
│<span class="w">                                                                      </span>bitscore<span class="o">)</span><span class="w">           </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span><span class="m">1</span><span class="o">]</span><span class="w">        </span>│
│<span class="w">    </span>--min-gene-len<span class="w">                                </span>INTEGER<span class="w">             </span>Minimum<span class="w"> </span>gene<span class="w"> </span>length<span class="w"> </span>│
│<span class="w">                                                                      </span><span class="k">for</span><span class="w"> </span>pyrodigal<span class="w"> </span><span class="o">(</span>nt<span class="o">)</span><span class="w">  </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span><span class="m">90</span><span class="o">]</span><span class="w">       </span>│
│<span class="w">    </span>--ttable<span class="w">                                      </span>INTEGER<span class="w">             </span>NCBI<span class="w"> </span>translation<span class="w">    </span>│
│<span class="w">                                                                      </span>table<span class="w"> </span><span class="k">for</span><span class="w"> </span>coding<span class="w">    </span>│
│<span class="w">                                                                      </span>sequences<span class="w">           </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span><span class="m">11</span><span class="o">]</span><span class="w">       </span>│
│<span class="w">    </span>--keep-proteins<span class="w">         </span>--no-keep-proteins<span class="w">                        </span>Keep<span class="w"> </span>the<span class="w"> </span>protein<span class="w">    </span>│
│<span class="w">                                                                      </span>FASTA<span class="w"> </span>used<span class="w"> </span><span class="k">for</span><span class="w">      </span>│
│<span class="w">                                                                      </span>searching<span class="w">           </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w">           </span>│
│<span class="w">                                                                      </span>no-keep-proteins<span class="o">]</span><span class="w">   </span>│
│<span class="w">    </span>--keep-domtbl<span class="w">           </span>--no-keep-domtbl<span class="w">                          </span>Keep<span class="w"> </span>raw<span class="w"> </span>domtblout<span class="w">  </span>│
│<span class="w">                                                                      </span>from<span class="w"> </span>pyHMMER<span class="w">        </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w">           </span>│
│<span class="w">                                                                      </span>keep-domtbl<span class="o">]</span><span class="w">        </span>│
│<span class="w">    </span>--combine-mode<span class="w">                                </span>TEXT<span class="w">                </span>How<span class="w"> </span>to<span class="w"> </span>combine<span class="w"> </span>hits<span class="w"> </span>│
│<span class="w">                                                                      </span>from<span class="w"> </span>multiple<span class="w"> </span>HMMs:<span class="w"> </span>│
│<span class="w">                                                                      </span>any<span class="p">|</span>all<span class="p">|</span>threshold<span class="w">   </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span>any<span class="o">]</span><span class="w">      </span>│
│<span class="w">    </span>--min-hmm-hits<span class="w">                                </span>INTEGER<span class="w">             </span>Minimum<span class="w"> </span>number<span class="w"> </span>of<span class="w">   </span>│
│<span class="w">                                                                      </span>HMMs<span class="w"> </span>that<span class="w"> </span>must<span class="w"> </span>hit<span class="w">  </span>│
│<span class="w">                                                                      </span>a<span class="w"> </span>contig<span class="w"> </span><span class="o">(</span><span class="k">for</span><span class="w">       </span>│
│<span class="w">                                                                      </span>threshold<span class="w"> </span>mode<span class="o">)</span><span class="w">     </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span><span class="m">1</span><span class="o">]</span><span class="w">        </span>│
│<span class="w">    </span>--save-target-pro…<span class="w">      </span>--no-save-target-…<span class="w">                        </span>Save<span class="w"> </span>matched<span class="w">        </span>│
│<span class="w">                                                                      </span>proteins<span class="w"> </span>per<span class="w"> </span>HMM<span class="w">    </span>│
│<span class="w">                                                                      </span>model<span class="w"> </span><span class="k">in</span><span class="w">            </span>│
│<span class="w">                                                                      </span>target_proteins/<span class="w">    </span>│
│<span class="w">                                                                      </span>subfolder<span class="w">           </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w">           </span>│
│<span class="w">                                                                      </span>no-save-target-pro…<span class="w"> </span>│
│<span class="w">    </span>--save-target-hmms<span class="w">      </span>--no-save-target-…<span class="w">                        </span>Build<span class="w"> </span>and<span class="w"> </span>save<span class="w"> </span>HMMs<span class="w"> </span>│
│<span class="w">                                                                      </span>from<span class="w"> </span>target<span class="w">         </span>│
│<span class="w">                                                                      </span>proteins<span class="w"> </span><span class="k">in</span><span class="w">         </span>│
│<span class="w">                                                                      </span>target_hmms/<span class="w">        </span>│
│<span class="w">                                                                      </span>subfolder<span class="w">           </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w">           </span>│
│<span class="w">                                                                      </span>no-save-target-hmm…<span class="w"> </span>│
│<span class="w">    </span>--hmm-mode<span class="w">                                    </span>TEXT<span class="w">                </span>HMM<span class="w"> </span>file<span class="w"> </span>type:<span class="w">       </span>│
│<span class="w">                                                                      </span><span class="s1">&#39;pure&#39;</span><span class="w"> </span><span class="o">(</span>one<span class="w"> </span>model<span class="w">   </span>│
│<span class="w">                                                                      </span>per<span class="w"> </span>file<span class="o">)</span><span class="w"> </span>or<span class="w">         </span>│
│<span class="w">                                                                      </span><span class="s1">&#39;mixed&#39;</span><span class="w">             </span>│
│<span class="w">                                                                      </span><span class="o">(</span>pressed/concatena…<span class="w"> </span>│
│<span class="w">                                                                      </span>HMMs<span class="o">)</span><span class="w">               </span>│
│<span class="w">                                                                      </span><span class="o">[</span>default:<span class="w"> </span>pure<span class="o">]</span><span class="w">     </span>│
│<span class="w">    </span>--help<span class="w">              </span>-h<span class="w">                                            </span>Show<span class="w"> </span>this<span class="w"> </span>message<span class="w">   </span>│
│<span class="w">                                                                      </span>and<span class="w"> </span>exit.<span class="w">           </span>│
╰──────────────────────────────────────────────────────────────────────────────────────────╯
</code></pre></div>
<p>Use <code>--output-folder</code> to change where the results are saved. The default is a folder called <code>phu-screen</code> in your current directory.</p>
<p>Use <code>--threads</code> to speed things up if you have multiple CPU cores available. This affects both the protein prediction step and the HMMER searches.</p>
<p>Use <code>--max-evalue</code> to make your searches more or less strict. The default is 1e-5, which is reasonably stringent. Lower values (like 1e-10) are more strict, while higher values (like 1e-3) are more permissive.</p>
<p>Use <code>--save-target-proteins</code> if you want to get the actual protein sequences from the contigs that matched each model. The saved proteins are taken only from contigs that passed final filtering and are grouped per-model (see "HMM modes" above).</p>
<p>Use <code>--save-target-hmms</code> to build custom HMM profiles from your target proteins. This works independently of <code>--save-target-proteins</code> - the tool can extract proteins temporarily just for HMM building if needed.</p>
<h2 id="examples">Examples<a class="headerlink" href="#examples" title="Permanent link">&para;</a></h2>
<p>Find contigs with any viral protein (default "any" preserves best-per-model hits):
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>assembly.fasta<span class="w"> </span>--combine-mode<span class="w"> </span>any<span class="w"> </span>viral_capsid.hmm<span class="w"> </span>viral_polymerase.hmm
</code></pre></div></p>
<p>Find contigs that have complete viral genomes (all four proteins):
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>contigs.fa<span class="w"> </span>--combine-mode<span class="w"> </span>all<span class="w"> </span>capsid.hmm<span class="w"> </span>portal.hmm<span class="w"> </span>primase.hmm<span class="w"> </span>terminase.hmm
</code></pre></div></p>
<p>Use multiple threads and save the matching proteins (per-model grouping depends on <code>--hmm-mode</code>):
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>large_assembly.fasta<span class="w"> </span>--threads<span class="w"> </span><span class="m">16</span><span class="w"> </span>--save-target-proteins<span class="w"> </span>marker.hmm
</code></pre></div></p>
<p>Be more strict about matches:
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>contigs.fa<span class="w"> </span>--max-evalue<span class="w"> </span>1e-10<span class="w"> </span>protein_family.hmm
</code></pre></div></p>
<p>Build custom HMMs from viral proteins (works with or without saving proteins):
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>viral_assembly.fasta<span class="w"> </span>--save-target-hmms<span class="w"> </span>--combine-mode<span class="w"> </span>all<span class="w"> </span>capsid.hmm<span class="w"> </span>polymerase.hmm
</code></pre></div></p>
<p>Complete viral screening workflow with custom HMM generation:
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>metagenome.fa<span class="w"> </span>--save-target-proteins<span class="w"> </span>--save-target-hmms<span class="w"> </span>--combine-mode<span class="w"> </span>threshold<span class="w"> </span>--min-hmm-hits<span class="w"> </span><span class="m">3</span><span class="w"> </span>viral_marker1.hmm<span class="w"> </span>viral_marker2.hmm<span class="w"> </span>viral_marker3.hmm<span class="w"> </span>viral_marker4.hmm
</code></pre></div></p>
<p>Screen with complex contig names (automatically handled):
<div class="highlight"><pre><span></span><code>phu<span class="w"> </span>screen<span class="w"> </span>--input-contigs<span class="w"> </span>scaffolds_with_complex<span class="p">|</span>names<span class="p">|</span>assembly.fa<span class="w"> </span>protein_family.hmm
</code></pre></div></p>
<h2 id="what-to-expect">What to expect<a class="headerlink" href="#what-to-expect" title="Permanent link">&para;</a></h2>
<p>Gene prediction usually takes 1-2 minutes per million base pairs of input. The pyHMMER searches are significantly faster than traditional HMMER due to in-memory processing and native Python threading. Performance scales well with the <code>--threads</code> option.</p>
<p>The output size depends on how many contigs match your criteria. In "any" mode, you might get quite a few contigs. In "all" mode, you'll typically get fewer but higher-confidence results.</p>
<p><strong>New performance benefits:</strong>
- <strong>Faster execution</strong>: pyHMMER eliminates subprocess overhead and file I/O
- <strong>Better memory usage</strong>: In-memory processing of all data
- <strong>No binary dependencies</strong>: Pure Python implementation
- <strong>Robust handling</strong>: Automatically deals with complex contig naming schemes</p>
<p>If you don't get any results, try relaxing your E-value threshold or check that your HMM files are in the correct format. If you get too many results, try using "all" mode instead of "any" mode, or make your E-value threshold more strict.</p>
<h2 id="requirements">Requirements<a class="headerlink" href="#requirements" title="Permanent link">&para;</a></h2>
<p>You need to have <strong>pyrodigal</strong>, <strong>pyHMMER</strong>, and <strong>seqkit</strong> installed and available. <strong>pyrodigal-gv</strong> is optional but recommended for viral genome analysis. Your input contigs should be in FASTA format, and your HMM files should be in HMMER3 format.</p>
<p><strong>Key improvements:</strong>
- <strong>No HMMER binary required</strong>: pyHMMER provides a pure Python implementation
- <strong>Automatic viral support</strong>: Uses pyrodigal-gv when available for viral gene prediction
- <strong>Robust contig handling</strong>: Automatically handles complex contig naming with multiple separators
- <strong>Flexible HMM building</strong>: Create custom profiles with or without saving intermediate proteins</p>
<p>The command expects DNA sequences as input, not protein sequences. If you already have predicted proteins, you should use pyHMMER directly rather than this command.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/camilogarciabotero/phu" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 2A10 10 0 0 0 2 12c0 4.42 2.87 8.17 6.84 9.5.5.08.66-.23.66-.5v-1.69c-2.77.6-3.36-1.34-3.36-1.34-.46-1.16-1.11-1.47-1.11-1.47-.91-.62.07-.6.07-.6 1 .07 1.53 1.03 1.53 1.03.87 1.52 2.34 1.07 2.91.83.09-.65.35-1.09.63-1.34-2.22-.25-4.55-1.11-4.55-4.92 0-1.11.38-2 1.03-2.71-.1-.25-.45-1.29.1-2.64 0 0 .84-.27 2.75 1.02.79-.22 1.65-.33 2.5-.33s1.71.11 2.5.33c1.91-1.29 2.75-1.02 2.75-1.02.55 1.35.2 2.39.1 2.64.65.71 1.03 1.6 1.03 2.71 0 3.82-2.34 4.66-4.57 4.91.36.31.69.92.69 1.85V21c0 .27.16.59.67.5C19.14 20.16 22 16.42 22 12A10 10 0 0 0 12 2"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://camilogarciabotero.github.io/phu/" target="_blank" rel="noopener" title="camilogarciabotero.github.io" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 21.5c-1.35-.85-3.8-1.5-5.5-1.5-1.65 0-3.35.3-4.75 1.05-.1.05-.15.05-.25.05-.25 0-.5-.25-.5-.5V6c.6-.45 1.25-.75 2-1 1.11-.35 2.33-.5 3.5-.5 1.95 0 4.05.4 5.5 1.5 1.45-1.1 3.55-1.5 5.5-1.5 1.17 0 2.39.15 3.5.5.75.25 1.4.55 2 1v14.6c0 .25-.25.5-.5.5-.1 0-.15 0-.25-.05-1.4-.75-3.1-1.05-4.75-1.05-1.7 0-4.15.65-5.5 1.5M12 8v11.5c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5V7c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5m1 3.5c1.11-.68 2.6-1 4.5-1 .91 0 1.76.09 2.5.28V9.23c-.87-.15-1.71-.23-2.5-.23q-2.655 0-4.5.84zm4.5.17c-1.71 0-3.21.26-4.5.79v1.69c1.11-.65 2.6-.99 4.5-.99 1.04 0 1.88.08 2.5.24v-1.5c-.87-.16-1.71-.23-2.5-.23m2.5 2.9c-.87-.16-1.71-.24-2.5-.24-1.83 0-3.33.27-4.5.8v1.69c1.11-.66 2.6-.99 4.5-.99 1.04 0 1.88.08 2.5.24z"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>